packages/sql/src/internal/client.ts

```diff
@@ -139,20 +143,15 @@ export function make({
         request.Request<SchemaError | E, A> & { i0: RA }
       >,
     ) =>
-    (
-      Resolver: RequestResolver.RequestResolver<any>,
+    <R = never>(
+      Resolver: RequestResolver.RequestResolver<any, R>,
       context = Context.empty() as Context.Context<any>,
     ) => {
-      const resolverWithSql = Effect.map(
-        Effect.serviceOption(TransactionConn),
-        _ =>
-          RequestResolver.provideContext(
-            Resolver,
-            Option.match(_, {
-              onNone: () => context,
-              onSome: tconn => Context.add(context, TransactionConn, tconn),
-            }),
-          ),
+      const resolverWithSql = Effect.contextWith<
+        R,
+        RequestResolver.RequestResolver<any, never>
+      >(_ =>
+        RequestResolver.provideContext(Resolver, Context.merge(_, context)),
```

Comment:
I couldn't think of an elegant way to correctly provide the dependencies for the resolver in an exclusive manner, so this should be looked at again. Now that schema supports `R`, this resolver has to be able to as well (without another wrapper from the outside world)

Feedback appreciated :)

Reply:
I'm thinking we side step returning an execute function if you require context for your schema's, and let the user decide how it will be provided to the resolver.

Otherwise you risk breaking batching if you supply a different context from different request sources.

Reply:
We can't detect whether the resolver needs or doesn't need dependencies in runtime, we can only lie in the type, omitting the `execute` function despite it existing in there.
Is that what you're suggesting?

Reply:
Ha! It's exactly what you're suggesting, ok 😆

Reply:
Yeah, I'm ok with a sneaky lie lol.